### DISCUSSION TITLE
Auth - public registration should triggers "users.create"  hooks

### DISCUSSION THREAD

**Discussion by gkielwasser** [directus/directus#9665] [Archive]
Hi,

I think the user that is created because of `AUTH_*_ALLOW_PUBLIC_REGISTRATION` should trigger appropriate hooks 'users.create' (filter/action).

Here is the [relevant part that creates the user](https://github.com/directus/directus/blob/86acc602e70f1440c9135ef2b077663ec0fcc0e9/api/src/auth/drivers/openid.ts#L139)

What do you think @rijkvanzanten  @aidenfoxx ?

Thanks

**Comment by aidenfoxx** [comment]
I believe it does? It creates a user via ItemService which should trigger the hook.

**Reply by gkielwasser** [reply]
> I believe it does? It creates a user via ItemService which should trigger the hook.

I checked again and indeed both events are well called.

In the "action" hook. I receive a null object for "accountability" that was skipping the next part of my flow. I think it makes sense as we don't have yet information about the current user.

Thanks !
